docs/structured-streaming-kafka-integration.md

```diff
@@ -325,6 +346,25 @@ The following configurations are optional:
   always pick up from where the query left off. Newly discovered partitions during a query will start at
   earliest.</td>
 </tr>
+<tr>
+  <td>endingOffsetsByTimestamp</td>
+  <td>json string
+  """ {"topicA":1000,"topicB":2000} """
+  </td>
+  <td>latest</td>
+  <td>batch query</td>
+  <td>The end point when a batch query is ended, a json string specifying an ending timesamp for each topic.
+  The returned offset for each partition is the earliest offset whose timestamp is greater than or equal to
+  the given timestamp in the corresponding partition. If the matched offset doesn't exist, the offset will
+  be set to latest.<p/>
+  <p/>
+  Spark simply passes the timestamp information to `KafkaConsumer.offsetForTimes`, and doesn't interpret or reason about the value. <p/>
+  For more details on `KafkaConsumer.offsetFormTimes`, please refer https://kafka.apache.org/21/javadoc/org/apache/kafka/clients/consumer/KafkaConsumer.html#offsetsForTimes-java.util.Map- for details.<p/>
+  Also the meaning of `timestamp` here can be vary according to Kafka configuration (`log.message.timestamp.type`): please refer https://kafka.apache.org/documentation/ for further details.<p/>
+  Note: This option requires Kafka 0.10.1.0 or higher.<p/>
+  Note2: `endingOffsetsByTimestamp` takes precedence over `endingOffsets`.
+  </td>
```

Comment:
Note3: For streaming queries it's not supported.

Reply:
The table already has a column which kind of queries are supported, and it's batch query for this option.

Reply:
```
    caseInsensitiveParams.get(ENDING_OFFSETS_BY_TIMESTAMP_OPTION_KEY).map(_ =>
      throw new IllegalArgumentException("ending timestamp not valid in streaming queries"))
```

Reply:
Please refer here: https://github.com/apache/spark/pull/23747/files/8434b2b493388e7c87b42bc7f137bf970f56ca09#diff-291798b0ad8f1d2c5c21dff52922e3d8R355

Reply:
Ahh, I see know `query type` set accordingly.